uint8 type
uint8[] data_field
